### PR TITLE
Added some dynamics to tabcontainer

### PR DESCRIPTION
**Pull request by Nortsx**
**Description**

Added a bit of dynamics to tab container, availability to add and remove tabs.

### CODE REVIEW

**Comment by andydotxyz** [comment]
Does this  fix #444?

**Comment by andydotxyz** [comment]
please run gofmt locally, it will fix the Travis issues

File: widget/tabcontainer.go [hunk 1]
```diff
@@ -163,6 +163,25 @@ func (t *TabContainer) buildTabBar(buttons []fyne.CanvasObject) *fyne.Container
...
 }
 
+
+// Append adds a new CanvasObject to the end of the group
+func (t *TabContainer) Append(item *TabItem) {
+	r := Renderer(t).(*tabContainerRenderer)
+	t.Items = append(t.Items, item)
+	r.tabBar.Objects = append(r.tabBar.Objects, t.makeButton(item))
+
+	t.CreateRenderer().Refresh()
```
Comment (andydotxyz) [inline widget/tabcontainer.go]: This is not correct - createRenderer creates a new renderer, then refreshes it. This is an orphan renderer.

Comment (andydotxyz) [reply widget/tabcontainer.go]: t.Refresh() should fix it

File: widget/tabcontainer.go [hunk 2]
```diff
@@ -163,6 +163,25 @@ func (t *TabContainer) buildTabBar(buttons []fyne.CanvasObject) *fyne.Container
 	return tabBar
 }
 
+
+// Append adds a new CanvasObject to the end of the group
+func (t *TabContainer) Append(item *TabItem) {
+	r := Renderer(t).(*tabContainerRenderer)
```
Comment (andydotxyz) [inline widget/tabcontainer.go]: Renderer() is deprecated.
If you need to get access to a renderer use cache.Renderer(t).

Comment (andydotxyz) [reply widget/tabcontainer.go]: p.s. can you also delete the old commented out code which this replaces? :)

File: widget/tabcontainer.go [hunk 3]
```diff
@@ -163,6 +163,25 @@ func (t *TabContainer) buildTabBar(buttons []fyne.CanvasObject) *fyne.Container
...
+	t.Items = append(t.Items, item)
+	r.tabBar.Objects = append(r.tabBar.Objects, t.makeButton(item))
+
+	t.CreateRenderer().Refresh()
+}
+
+// Remove tab by index
+func (t *TabContainer) Remove(index int) {
```
Comment (andydotxyz) [inline widget/tabcontainer.go]: To be consistent with the methods above it should be RemoveIndex if you are passing an int.
Remove() would take a *TabItem.

We are trying to ensure consistency to all public API, sorry for the nit-picking

Comment (Nortsx) [reply widget/tabcontainer.go]: Done:)

File: widget/tabcontainer.go (hunk 2 above)
Comment (Nortsx) [reply widget/tabcontainer.go]: Done:)

File: widget/tabcontainer.go (hunk 1 above)
Comment (Nortsx) [reply widget/tabcontainer.go]: Yup, sorry, got a wrong way here:)

File: widget/tabcontainer.go [hunk 4]
```diff
@@ -163,10 +144,42 @@ func (t *TabContainer) buildTabBar(buttons []fyne.CanvasObject) *fyne.Container
...
+	t.Refresh()
+}
+
+// Remove tab by value
+func (t *TabContainer) Remove(item *TabItem) {
+	r := cache.Renderer(t).(*tabContainerRenderer)
+
+	for index, existingItem := range t.Items {
+		if existingItem == item {
+			t.Items = append(t.Items[:index], t.Items[index+1:]...)
```
Comment (andydotxyz) [inline widget/tabcontainer.go]: Would it not be a little neater to call `RemoveIndex(index)` here to remove the duplication?

File: widget/tabcontainer.go [hunk 5]
```diff
@@ -163,10 +144,42 @@ func (t *TabContainer) buildTabBar(buttons []fyne.CanvasObject) *fyne.Container
 	return tabBar
 }
 
+// Append adds a new CanvasObject to the end of the group
```
Comment (andydotxyz) [inline widget/tabcontainer.go]: This looks like a copy paste issue. The docs seem slightly terse but apart from this line they are correct ;)

**Pull request merged**